include/ry/runtime_net_utils.hpp

```
@@ -0,0 +1,198 @@
+    char port_str[16];
+    snprintf(port_str, sizeof(port_str), "%lld", (long long)port);
+
+    *out = nullptr;
+    if (::getaddrinfo(host, port_str, &hints, out) != 0)
+        return -1;
+    return 0;
+}
+
+// ===== SSRF protection =====
+
+static inline bool ry_net_is_private_ipv4_raw(uint32_t ip) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)